client/messaging/action-creators.ts

```diff
@@ -0,0 +1,21 @@
+import { MouseEvent } from 'react'
+import { openDialog } from '../dialogs/action-creators'
+import { DialogType } from '../dialogs/dialog-type'
+import { ThunkAction } from '../dispatch-registry'
+
+export function maybeOpenExternalLinkDialog(e: MouseEvent<HTMLAnchorElement>): ThunkAction {
+  return (dispatch, getState) => {
+    const trustedDomains: string[] = getState().settings.local.trustedDomains
```

Comment:
I believe TypeScript can figure out the type here automatically, so you don't have to explicitly type it.

Reply:
Not really, tec27 mentioned it [here](https://github.com/ShieldBattery/ShieldBattery/blob/master/client/ladder/action-creators.ts#L11). Moreover, even if `getState()` is typed, `local` does not show me its fields.